iceoryx_posh/include/iceoryx_posh/internal/mepoo/segment_manager.hpp

```diff
@@ -57,7 +58,7 @@ class SegmentManager
                        uint64_t size,
                        bool isWritable,
                        uint64_t segmentId,
-                       const iox::mepoo::MemoryInfo& memoryInfo = iox::mepoo::MemoryInfo())
+                       const iox::mepoo::MemoryInfo& memoryInfo = iox::mepoo::MemoryInfo()) noexcept
```

Comment:
Please move implementation in inl file.

Reply:
Will do this in the follow-up as I would like to do some more refactoring.

Reply:
To satisfy this checkbox: All open points are addressed and tracked via issues. Could you please link the issue where this open point is tracked.

Reply:
Already tracked in #722